### ISSUE TITLE
/.singularity.d/runscript: 3: exec: trim-low-abund.py: not found

### ISSUE THREAD

**Issue by yuzie0314** [sourmash-bio/sourmash#2606]
Hi the creator of sourmash, 

Recently we used your tool to annotate our reads, and we think that we need to trim low abundance reads inside our illumina reads according to your tutorial. We want to use singularity to build the requirement but when we test on the command `singularity run sourmash.4.6.1.sif trim-low-abund.py`, which show that trim-low-abund.py not found. Do you have any idea?
Even we tested on our conda environment, we failed (installed by the command `conda install -c bioconda sourmash`).

P.S. I also found [this sourmash image](https://singularityhub.github.io/shpc-registry/r/quay.io-biocontainers-sourmash/), but didn't know how to make use of this image. (I am not really familliar with singularity, sorry)

thanks in advance,
Yuzie

**Comment by ctb** [comment]
hi @yuzie0314,

tl;dr trim-low-abund is from the khmer package and is no longer supported. It's OK to do with out.

Longer: 

* we no longer support the software package containing trim-low-abund, I'm afraid!
* however, we have been rethinking read trimming for most purposes: see [this issue](https://github.com/dib-lab/genome-grist/issues/141) for more details.
* so it's ok not to do it, especially since there's not really an alternative!

If you are set on doing it, I believe you can install the khmer software in a few different ways and I can get back to you on that - just let me know you want that.

**Comment by yuzie0314** [comment]
Dear ctb the author of sourmash (sm),
Sorry for the late, I was busy with my proposal these days. 
We don't know whether trim low abundance before compare the signatures from sm is more efficient than not filtering reads. Our problem is we have to find whether the reference genome contents shown in our reads, and if we use full reads from a deeply sequencing sample, which will take longer to complete the comparison. We intend to use sm to help us quickly identify the reference geomes that present in our sample (reads) and speed up the downstream analysis.

I don't know whether this message help you to understand my problem, but really thanks for your time to elaborate the trim-low-abund.py script.

Many thanks,
Yuzie

**Comment by ctb** [comment]
sure - two specific responses:

* the k-mer based trimming done by trim-low-abund is pretty slow! so it is unlikely to speed up your pipeline.
* sourmash's performance will not be substantially improved by the trimming.

So my advice is to simply ignore our k-mer trimming :).

**Comment by yuzie0314** [comment]
Thanks for your advice,
Agree with your opinions.
